packages/matchbox/src/components/Pagination/Pagination.js

```diff
@@ -1,119 +1,71 @@
+import { margin } from 'styled-system';
+import { createPropTypes } from '@styled-system/prop-types';
+import { tokens } from '@sparkpost/design-tokens';
 
+import { Box } from '../Box';
 import { Button, buttonsFrom } from '../Button';
 import { Pager } from '../Pager';
 import { MoreHoriz } from '@sparkpost/matchbox-icons';
-import styles from './Pagination.module.scss';
-
-class Pagination extends Component {
-  static displayName = 'Pagination';
-
-  static propTypes = {
-    /**
-     * Sets the current page number
-     */
-    currentPage: PropTypes.number,
-
-    /**
-     * The total number of pages
-     */
-    pages: PropTypes.number.isRequired,
-
-    /**
-     * The number of page buttons to display. Odd numbers look better.
-     */
-    pageRange: PropTypes.number.isRequired,
-
-    /**
-     * Callback when page is changed. Index passed as argument.
-     */
-    onChange: PropTypes.func,
-
-    /**
-     * Hides first and last buttons
-     */
-    marginsHidden: PropTypes.bool,
-
-    /**
-     * Flat style buttons
-     */
-    flat: PropTypes.bool,
-
-    /**
-     * Selected page color
-     */
-    selectedColor: PropTypes.oneOf(['orange', 'blue', 'navy', 'purple', 'red']),
-  };
-
-  static defaultProps = {
-    currentPage: 1,
-    selectedColor: 'orange',
-  };
-
-  state = {
-    index: 0,
-    hasPrevious: false,
-    hasNext: true,
-  };
 
-  componentDidMount() {
-    this.handlePageChange(this.props.currentPage - 1);
+const StyledMoreHoriz = styled(MoreHoriz)`
+  fill: ${tokens.color_gray_500};
+`;
+
+function Pagination(props) {
+  const [index, setIndex] = React.useState(0);
+  const [hasPrevious, setHasPrevious] = React.useState(false);
+  const [hasNext, setHasNext] = React.useState(false);
+
+  const {
+    currentPage,
+    pages,
+    pageRange,
+    onChange,
+    marginsHidden,
+    flat,
+    selectedColor,
+    ...rest
+  } = props;
+
+  function handlePageChange(index) {
+    setIndex(index);
+    setHasPrevious(index > 0);
+    setHasNext(index < pages - 1);
+
+    onChange && onChange(index);
   }
 
-  componentDidUpdate(prevProps) {
-    const { pages, pageRange, currentPage } = this.props;
-    if (
-      pages !== prevProps.pages ||
-      pageRange !== prevProps.pageRange ||
-      currentPage !== prevProps.currentPage
-    ) {
-      this.handlePageChange(currentPage - 1);
-    }
+  function handleNext() {
+    handlePageChange(index + 1);
   }
 
-  handlePageChange(index) {
-    this.setState({
-      index,
-      hasPrevious: index > 0,
-      hasNext: index < this.props.pages - 1,
-    });
-
-    this.props.onChange && this.props.onChange(index);
+  function handlePrevious() {
+    handlePageChange(index - 1);
   }
 
-  handleNext = () => {
-    this.handlePageChange(this.state.index + 1);
-  };
-
-  handlePrevious = () => {
-    this.handlePageChange(this.state.index - 1);
-  };
-
-  _createButtons(start, end) {
+  function _createButtons(start, end) {
```

Comment:
This is a really good opportunity to try out `React.useMemo` if you haven't used it yet. I can walk you through it if you'd like, but it is essentially the same as `useEffect` except you return a value instead. Great for functions like.